### PR TITLE
Rails5 upgrade

### PR DESCRIPTION
**Pull request by chunkzer**

### CODE REVIEW

File: lib/daylight/associations.rb [hunk 1]
```diff
@@ -92,9 +92,10 @@ def has_many name, options={}
     # ActiveResource::Associations#belongs_to
 
     def belongs_to name, options={}
+
       create_reflection(:belongs_to, name, options).tap do |reflection|
 
-        nested_attribute_key = "#{reflection.name}_attributes"
+        nested_attribute_key = "#{reflection.name}"
```
Comment (manfrin) [inline lib/daylight/associations.rb:98]: explain?

File: lib/daylight/collection.rb [hunk 2]
```diff
@@ -109,7 +109,7 @@ def create_resource attributes={}
 
       ##
       # Performs the work of merging known attributes to the supplied +attributes+
-      # on the resource, setting the +prefix_options+ to the known params and and
+      # on the resource, setting the +prefix_options+ to the known params and
```
Comment (manfrin) [inline lib/daylight/collection.rb:112]: hah

File: rails/daylight/refiners.rb [hunk 3]
```diff
@@ -126,7 +127,15 @@ def reflection_names
     # Supplies where conditions and returns the resulting +ActiveRecord::Relation+.
     # Raises +ArgumentError+ if the keys are not valid attributes on the model.
     def filter_by params
-      where (params||{}).with_indifferent_access.assert_valid_keys(attribute_names + reflection_names)
+      params = params || {}
+      dummy = self.new(params)
+      params = params.with_indifferent_access rescue params.to_h
+      params.each do |k, v|
+        if dummy[k].class == TrueClass || dummy[k].class == FalseClass
```
Comment (manfrin) [inline rails/daylight/refiners.rb:134]: why would you check for false?

File: rails/extensions/nested_attributes_ext.rb [hunk 4]
```diff
@@ -127,7 +127,7 @@ def has_many_type(association)
       reflection = association.reflection
       if reflection.try(:has_and_belongs_to_many?) ||
          (reflection.try(:parent_reflection) &&
-          reflection.parent_reflection.last.try(:macro) == :has_and_belongs_to_many)
+          reflection.parent_reflection.try(:macro) == :has_and_belongs_to_many)
```
Comment (manfrin) [inline rails/extensions/nested_attributes_ext.rb:130]: this is an actual change...?

Comment (chunkzer) [reply rails/extensions/nested_attributes_ext.rb:130]: Yeah, parent_reflection before returned results as part of an array but no more.

File: rails/daylight/refiners.rb (hunk 3 above)
Comment (chunkzer) [reply rails/daylight/refiners.rb:134]: This one was pretty painful and this block of code is a workaround to that issue. 
When working from GFE I had the issue were daylight was not returning the right stuff for queries such as. Merchant.where(active: "false") or Merchant.where(active: "1"). Effectively, anything that wasn't 0 or 1 evaluated to false for purposes of the AR query.

File: lib/daylight/collection.rb (hunk 2 above)
Comment (chunkzer) [reply lib/daylight/collection.rb:112]: Every grain of sand...

File: rails/extensions/request_id_ext.rb [hunk 5]
```diff
@@ -5,17 +5,22 @@
...
 # See
 # ActionDispatch::RequestId
-
 module Daylight::RequestIdExt
+  def call(env)
+    req = ActionDispatch::Request.new env
+    if env["HTTP_X_REQUEST_ID"].presence
+      req.request_id = x_make_request_id(req.x_request_id)
+    else
+      req.request_id = make_request_id(req.x_request_id)
```
Comment (manfrin) [inline rails/extensions/request_id_ext.rb:14]: why is this making a req only to set one attribute and then use that attribute later -- why make this req object at all?

File: lib/daylight/associations.rb (hunk 1 above)
Comment (chunkzer) [reply lib/daylight/associations.rb:98]: Created reflections don't have the "_attributes" suffix anymore. We were just sending a "merchant" over "merchant_attributes" so we failed everytime.

File: rails/extensions/request_id_ext.rb (hunk 5 above)
Comment (chunkzer) [reply rails/extensions/request_id_ext.rb:14]: IIRC It's a monkey patch on ActionDispatch::RequestID, so Daylight is setting up some headers the way it likes it.

File: rails/daylight/refiners.rb (hunk 3 above)
Comment (manfrin) [reply rails/daylight/refiners.rb:134]: but you're only mutating params if they are `TrueClass` or `FalseClass`, and in both of those cases, it'll end up the same.

File: rails/extensions/request_id_ext.rb (hunk 5 above)
Comment (manfrin) [reply rails/extensions/request_id_ext.rb:14]: But you're not using req here at all, except as a temporary holder for request_id.

Comment (chunkzer) [reply rails/extensions/request_id_ext.rb:14]: @app.call(env).tap { |_status, headers, _body| headers[ActionDispatch::RequestId::X_REQUEST_ID] = req.request_id }

I'm using it here

File: rails/daylight/refiners.rb (hunk 3 above)
Comment (chunkzer) [reply rails/daylight/refiners.rb:134]: I'm checking the "booleanity" of the model's attributes. If it's a boolean model I need to cast the attribute value into a boolean which normally happens automagically afaik but these issue was persistent so this manual casting was my fix.

Comment (manfrin) [reply rails/daylight/refiners.rb:134]: but you're not casting anything -- if it's `1`, then it wont match on `== TrueClass`, if the param is `true`, then you cast it to `true`. In the end, nothing changes.

File: rails/extensions/request_id_ext.rb (hunk 5 above)
Comment (manfrin) [reply rails/extensions/request_id_ext.rb:14]: right -- so why set it on req in the first place, why do you need the request model?

**Pull request merged**